### PR TITLE
Crop Lab: Implement Settings for anchor points

### PR DESCRIPTION
**Pull request by initialshl**
Related to #1220 

Created Crop Lab Settings dialog.
Anchor point is implemented for Crop To Aspect Ratio.

croplab-settings

### CODE REVIEW

**Comment by initialshl** [comment]
> An option is to put crop lab as a group on its own in the ribbon so that you have a bottom right arrow to crop lab

I'm not sure if users use crop lab often enough to warrant it to be expanded on the ribbon, this may require more dogfooding and feedback.
Personally, I think that we should make crop lab and its settings open in a side pane (not sure if we have enough time to), so that the user can crop all he wants using one mouse click each time.

**Comment by damithc** [comment]
> I'm not sure if users use crop lab often enough to warrant it to be expanded on the ribbon, this may require more dogfooding and feedback.

Most likely we'll be collapsing each lab to just one button on the ribbon, to save ribbon space. Users can use the 'add to quick access bar' feature to add frequently used features to the quick access bar.

> Personally, I think that we should make crop lab and its settings open in a side pane (not sure if we have enough time to), so that the user can crop all he wants using one mouse click each time.

If we do that, it should be an option rather than the only way to use crop lab.  Otherwise those who need to crop something just once will have to do too many things i.e. open panel, click, close panel.

File: PowerPointLabs/PowerPointLabs/CropLab/CropLabSettings.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        Middle,
+        MiddleRight,
+        BottomLeft,
+        Bottom,
+        BottomRight
+    }
+
+    internal static class CropLabSettings
+    {
+        private static AnchorPosition anchorPosition = AnchorPosition.Middle;
```
Comment (jamos-tay) [inline PowerPointLabs/PowerPointLabs/CropLab/CropLabSettings.cs]: Why are there 2 vars anchorPostition and AnchorPosition? Try `public static AnchorPosition AnchorPosition { get; set; } = AnchorPosition.Middle;`

File: PowerPointLabs/PowerPointLabs/CropLab/CropLabSettingsDialog.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    {
+        private AnchorPosition selectedAnchor;
+
+        public CropLabSettingsDialog()
+        {
+            selectedAnchor = CropLabSettings.AnchorPosition;
+            InitializeComponent();
+        }
+
+        public AnchorPosition SelectedAnchor
```
Comment (jamos-tay) [inline PowerPointLabs/PowerPointLabs/CropLab/CropLabSettingsDialog.xaml.cs]: Same as above

File: PowerPointLabs/PowerPointLabs/CropLab/CropLabSettings.cs (hunk 1 above)
Comment (initialshl) [inline PowerPointLabs/PowerPointLabs/CropLab/CropLabSettings.cs]: Thanks, fixed

File: PowerPointLabs/PowerPointLabs/CropLab/CropLabSettingsDialog.xaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            <RadioButton x:Name="bottomAnchorRadioButton" Content="" HorizontalAlignment="Left" Margin="10,6,0,0" VerticalAlignment="Top"
+                         IsChecked="{Binding Path=SelectedAnchor, Converter={StaticResource EnumToBooleanConverter}, ConverterParameter={x:Static local:AnchorPosition.Bottom}}" />
+            <RadioButton x:Name="bottomRightAnchorRadioButton" Content="" HorizontalAlignment="Left" Margin="10,6,0,0" VerticalAlignment="Top"
+                         IsChecked="{Binding Path=SelectedAnchor, Converter={StaticResource EnumToBooleanConverter}, ConverterParameter={x:Static local:AnchorPosition.BottomRight}}" />
+        </WrapPanel>
+        <Button x:Name="okButton" Content="OK" HorizontalAlignment="Right" Margin="0,0,100,10" VerticalAlignment="Bottom" Width="80" IsDefault="True" Click="OkButton_Click"/>
+        <Button x:Name="cancelButton" Content="Cancel" HorizontalAlignment="Right" Margin="0,0,10,10" VerticalAlignment="Bottom" Width="80" IsCancel="True" Click="CancelButton_Click"/>
```
Comment (Kenneth-LJS) [inline PowerPointLabs/PowerPointLabs/CropLab/CropLabSettingsDialog.xaml]: Apply the property
IsDefault="True" on okButton, and
IsCancel="True" on cancelButton

File: PowerPointLabs/PowerPointLabs/CropLab/CropLabSettingsDialog.xaml.cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        private void OkButton_Click(object sender, RoutedEventArgs e)
+        {
+            CropLabSettings.AnchorPosition = SelectedAnchor;
+            Close();
+        }
+
+        private void CancelButton_Click(object sender, RoutedEventArgs e)
+        {
+            Close();
+        }
```
Comment (Kenneth-LJS) [inline PowerPointLabs/PowerPointLabs/CropLab/CropLabSettingsDialog.xaml.cs]: CancelButton_Click can be removed after setting IsCancel="True" on the cancel button.

**Pull request merged**